chemprop/nn/metrics.py

```diff
@@ -63,160 +77,473 @@ def forward(
         lt_mask: Tensor | None = None,
         gt_mask: Tensor | None = None,
     ):
+        """Calculate the mean loss function value given predicted and target values
+
+        Parameters
+        ----------
+        preds : Tensor
+            a tensor of shape `b x t x u` (regression), `b x t` (binary classification), or
+            `b x t x c` (multiclass classification) containing the predictions, where `b` is the
+            batch size, `t` is the number of tasks to predict, `u` is the number of
+            targets to predict for each task, and `c` is the number of classes.
+        targets : Tensor
+            a float tensor of shape `b x t` containing the target values
+        mask : Tensor
+            a boolean tensor of shape `b x t` indicating whether the given prediction should be
+            included in the loss calculation
+        weights : Tensor
+            a tensor of shape `b` or `b x 1` containing the per-sample weight
+        lt_mask: Tensor
+        gt_mask: Tensor
+
+        Returns
+        -------
+        Tensor
+            a scalar containing the fully reduced loss
+        """
```

Comment:
1. add  `-> None`
2. remove the docstring related to return
```suggestion
    ) -> None:
        """Calculate the mean loss function value given predicted and target values

        Parameters
        ----------
        preds : Tensor
            a tensor of shape `b x t x u` (regression), `b x t` (binary classification), or
            `b x t x c` (multiclass classification) containing the predictions, where `b` is the
            batch size, `t` is the number of tasks to predict, `u` is the number of
            targets to predict for each task, and `c` is the number of classes.
        targets : Tensor
            a float tensor of shape `b x t` containing the target values
        mask : Tensor
            a boolean tensor of shape `b x t` indicating whether the given prediction should be
            included in the loss calculation
        weights : Tensor
            a tensor of shape `b` or `b x 1` containing the per-sample weight
        lt_mask: Tensor
        gt_mask: Tensor
        """
```